src/Microsoft.SqlTools.ServiceLayer/ObjectManagement/ObjectTypes/Database/DatabaseHandler.cs

```diff
@@ -26,17 +87,507 @@ public override bool CanHandleType(SqlObjectType objectType)
+
+                            if (prototype.NumberOfLogFiles > 0)
+                            {
+                                var logFile = prototype.Files[1];
+                                Debug.Assert(dataFile.DatabaseFileType == FileType.Log, "Expected first database file to be a log file for new database prototype");
+                                logFile.Name = $"{sanitizedName}_log";
+                            }
+                        }
+
+                        if (database.Owner != null && database.Owner != defaultValue)
+                        {
+                            prototype.Owner = database.Owner;
+                        }
+                        if (database.CollationName != null && database.CollationName != defaultValue)
+                        {
+                            prototype.Collation = database.CollationName;
+                        }
+                        if (database.RecoveryModel != null)
+                        {
+                            prototype.RecoveryModel = recoveryModelEnums[database.RecoveryModel];
+                        }
+                        if (database.CompatibilityLevel != null)
+                        {
+                            prototype.DatabaseCompatibilityLevel = compatLevelEnums[database.CompatibilityLevel];
+                        }
+                        if (prototype is DatabasePrototype110 db110 && database.ContainmentType != null)
+                        {
+                            db110.DatabaseContainmentType = containmentTypeEnums[database.ContainmentType];
+                        }
+
+                        string sqlScript = string.Empty;
+                        using (var actions = new DatabaseActions(dataContainer, configAction, prototype))
+                        using (var executionHandler = new ExecutonHandler(actions))
+                        {
+                            executionHandler.RunNow(runType, this);
+                            if (executionHandler.ExecutionResult == ExecutionMode.Failure)
+                            {
+                                throw executionHandler.ExecutionFailureException;
+                            }
+
+                            if (runType == RunType.ScriptToWindow)
+                            {
+                                sqlScript = executionHandler.ScriptTextFromLastRun;
+                            }
+                        }
+
+                        return sqlScript;
+                    }
+                }
+                finally
+                {
+                    ServerConnection serverConnection = dataContainer?.Server?.ConnectionContext;
+                    if (serverConnection != null && serverConnection.IsOpen)
+                    {
+                        serverConnection.Disconnect();
+                    }
+                }
+            }
+        }
+
+        private static string SanitizeFileName(string name)
```

Comment:
Doc comments for helper functions are useful - there's a lot of different things "sanitize" could mean, so having the comment explain what it's sanitizing it for use for would be helpful.